apis/vshn/v1/dbaas_vshn_postgresql.go

```diff
@@ -74,6 +74,24 @@ type VSHNPostgreSQLParameters struct {
 
 	// Encryption contains settings to control the storage encryption of an instance.
 	Encryption VSHNPostgreSQLEncryption `json:"encryption,omitempty"`
+
+	// UpdateStrategy indicates how and when updates the the instance spec will be applied.
+	UpdateStrategy VSHNPostgreSQLUpdateStrategy `json:"updateStrategy,omitempty"`
+}
+
+const VSHNPostgreSQLUpdateStrategyTypeOnChange = "OnChange"
+const VSHNPostgreSQLUpdateStrategyTypeOnDelete = "OnDelete"
+
+// VSHNPostgreSQLUpdateStrategy indicates how and when updates the the instance spec will be applied.
+type VSHNPostgreSQLUpdateStrategy struct {
+	// +kubebuilder:validation:Enum="OnDelete";"OnChange"
+	// +kubebuilder:default="OnDelete"
```

Comment:
I'm not entirely sure which behavior should be the default. What do you think?

Reply:
I would definitely go with the safest route which is OnDelete. We do not want to surprise the customer.

Reply:
Yeah, but that changing the disk size doesn't apply until maintenance is also a surprise :sweat_smile:

Reply:
I'd actually argue to go with the `OnChange` method by default. As it's how vanilla k8s does things as well and should be less of a surprise for users.